ui/src/tempVars/components/TagValuesTemplateBuilder.tsx

```diff
@@ -0,0 +1,308 @@
+import React, {PureComponent} from 'react'
+import _ from 'lodash'
+
+import {
+  TemplateBuilderProps,
+  TemplateValueType,
+  RemoteDataState,
+} from 'src/types'
+import {ErrorHandling} from 'src/shared/decorators/errors'
+import Dropdown from 'src/shared/components/Dropdown'
+import {showDatabases, showMeasurements} from 'src/shared/apis/metaQuery'
+import {proxy} from 'src/utils/queryUrlGenerator'
+import parseShowDatabases from 'src/shared/parsing/showDatabases'
+import parseShowMeasurements from 'src/shared/parsing/showMeasurements'
+import parseShowTagValues from 'src/shared/parsing/showTagValues'
+import {fetchTagKeys} from 'src/tempVars/components/TagKeysTemplateBuilder'
+import TemplateMetaQueryPreview from 'src/tempVars/components/TemplateMetaQueryPreview'
+import DropdownPlaceholder from 'src/shared/components/DropdownPlaceholder'
+
+interface State {
+  databases: string[]
+  databasesStatus: RemoteDataState
+  selectedDatabase: string
+  measurements: string[]
+  measurementsStatus: RemoteDataState
+  selectedMeasurement: string
+  tagKeys: string[]
+  tagKeysStatus: RemoteDataState
+  selectedTagKey: string
+  tagValues: string[]
+  tagValuesStatus: RemoteDataState
+}
+
+@ErrorHandling
+export default class KeysTemplateBuilder extends PureComponent<
+  TemplateBuilderProps,
+  State
+> {
+  constructor(props) {
+    super(props)
+
+    const selectedDatabase = _.get(props, 'template.query.db', '')
+    const selectedMeasurement = _.get(props, 'template.query.measurement', '')
+    const selectedTagKey = _.get(props, 'template.query.tagKey', '')
+
+    this.state = {
+      databases: [],
+      databasesStatus: RemoteDataState.Loading,
+      selectedDatabase,
+      measurements: [],
+      measurementsStatus: RemoteDataState.Loading,
+      selectedMeasurement,
+      tagKeys: [],
+      tagKeysStatus: RemoteDataState.Loading,
+      selectedTagKey,
+      tagValues: [],
+      tagValuesStatus: RemoteDataState.Loading,
+    }
+  }
+
+  public async componentDidMount() {
```

Comment:
very cool.

might this (and previous components that are doing this pattern) merit a `try/catch`?